spark3/src/main/java/org/apache/iceberg/spark/Spark3Util.java

```diff
@@ -292,6 +292,12 @@ public Transform hour(String sourceName, int sourceId) {
             return Expressions.hours(sourceName);
           }
 
+          @Override
+          public Transform alwaysNull(int fieldId, String sourceName, int sourceId) {
+            // do nothing for alwaysNull, it doesn't need to be converted to a transform
+            return null;
```

Comment:
Instead of returning null here and discarding nulls automatically, I think this should return null (or some other signaling value) and fix the part that consumes the output of the visitor.

Reply:
👌   Updated the PR and `Spark3Util` returns `null` to avoid create dummy objects.